tutorials/07-implementing-a-physics-plugin.md

```diff
@@ -0,0 +1,174 @@
+cd simple_plugin
+```
+
+Then download the example loader into your current directory by:
+
+```bash
+wget https://raw.githubusercontent.com/ignitionrobotics/ign-physics/main/examples/hello_world_plugin/HelloWorldPlugin.cc
+```
+
+### Examine the code
+
+We first include these:
+- \ref ignition::physics::FeatureList "FeatureList" header: to define a list of
+\ref ignition::physics::Feature "Feature" that will be implemented for the
+plugin, e.g. \ref ignition::physics::GetEngineInfo "GetEngineInfo".
+- \ref ignition::physics::FeaturePolicy "FeaturePolicy" header: to specify the
+metadata about the coordinate system (e.g. 2 or 3 dimensions) and numeric system
+(e.g. float, int) that the `Features` will be implemented accordingly.
+- `GetEntities.hh` header: to retrieve the pre-defined list of features in
+Ignition Physics. Please refer to this [API Reference](https://ignitionrobotics.org/api/physics/3.0/GetEntities_8hh.html)
+for a list of features defined in `GetEntities.hh` and \ref physicsplugin
+"Understanding the Physics Plugin" tutorial for an example list of common features
+implemented with a plugin.
+- `Register.hh` header: to register a plugin to Ignition Physics.
+
+```cpp
+#include <ignition/physics/FeatureList.hh>
+#include <ignition/physics/FeaturePolicy.hh>
+#include <ignition/physics/GetEntities.hh>
+#include <ignition/physics/Register.hh>
+```
+
+Next, let us define a namespace `mock` and implement our plugin in it.
+We will add a pre-defined feature `GetEngineInfo` to our plugin's
+`HelloWorldFeatureList` as follow:
+
+```cpp
+struct HelloWorldFeatureList : ignition::physics::FeatureList<
+    ignition::physics::GetEngineInfo
+> { };
+```
+
+The plugin will be able to return its physics engine metadata.
+We will now implement our plugin class named `HelloWorldPlugin`
+using the defined `FeatureList` above. The class is inherited from
+\ref ignition::physics::Implements3d "Implements3d" to declare that our plugin's
+`HelloWorldFeatureList` will be in the 3D coordinate system.
+
+```cpp
+class HelloWorldPlugin
+    : public ignition::physics::Implements3d<HelloWorldFeatureList>
+{
+  using Identity = ignition::physics::Identity;
+
+  public: Identity InitiateEngine(std::size_t /*_engineID*/) override
+  {
+    this->engineName = "HelloWorld";
+
+    return this->GenerateIdentity(0);
+  }
+
+  public: std::size_t GetEngineIndex(const Identity &/*_id*/) const override
+  {
+    return 0;
+  }
+
+  public: const std::string &GetEngineName(const Identity &/*_id*/) const override
+  {
+    return this->engineName;
+  }
+
+  std::string engineName;
+};
+```
+
+For now, because we do not have any real physics engines, we will define a dummy
+physics engines inside member function `InitiateEngine` by simply setting
+```cpp
+this->engineName = "HelloWorld"
+```
+and returning the engine object using \ref ignition::physics::Identity. Then, we
+define the metadata getters `GetEngineIndex` and `GetEngineName` for our plugin
+and that's it.
```

Comment:
would be nice to point out where `GetEngineIndex` and `GetEngineName` features come from, and what are some other features available. We can point to https://github.com/ignitionrobotics/ign-physics/blob/main/tutorials/03_physics_plugins.md#feature-comparison for some other example feature names, and https://github.com/ignitionrobotics/ign-physics/tree/main/include/ignition/physics for implemented`Features` in `ign-physics`

Reply:
I did add the pointing in L37-41, but a repeat here could also help.